astropy/io/fits/tests/test_image.py

```diff
@@ -516,125 +516,28 @@ def test_section_data_cube(self):
-        assert (d.section[0:1, 1:2, 0:3] == dat[0:1, 1:2, 0:3]).all()
-        assert (d.section[0:1, 1:2, 1:2] == dat[0:1, 1:2, 1:2]).all()
-        assert (d.section[0:1, 1:2, 1:3] == dat[0:1, 1:2, 1:3]).all()
-        assert (d.section[0:1, 1:2, 2:3] == dat[0:1, 1:2, 2:3]).all()
-        assert (d.section[0:1, 1:3, 0:1] == dat[0:1, 1:3, 0:1]).all()
-        assert (d.section[0:1, 1:3, 0:2] == dat[0:1, 1:3, 0:2]).all()
-        assert (d.section[0:1, 1:3, 0:3] == dat[0:1, 1:3, 0:3]).all()
-        assert (d.section[0:1, 1:3, 1:2] == dat[0:1, 1:3, 1:2]).all()
-        assert (d.section[0:1, 1:3, 1:3] == dat[0:1, 1:3, 1:3]).all()
-        assert (d.section[0:1, 1:3, 2:3] == dat[0:1, 1:3, 2:3]).all()
-        assert (d.section[1:2, 0:1, 0:1] == dat[1:2, 0:1, 0:1]).all()
-        assert (d.section[1:2, 0:1, 0:2] == dat[1:2, 0:1, 0:2]).all()
-        assert (d.section[1:2, 0:1, 0:3] == dat[1:2, 0:1, 0:3]).all()
-        assert (d.section[1:2, 0:1, 1:2] == dat[1:2, 0:1, 1:2]).all()
-        assert (d.section[1:2, 0:1, 1:3] == dat[1:2, 0:1, 1:3]).all()
-        assert (d.section[1:2, 0:1, 2:3] == dat[1:2, 0:1, 2:3]).all()
-        assert (d.section[1:2, 0:2, 0:1] == dat[1:2, 0:2, 0:1]).all()
-        assert (d.section[1:2, 0:2, 0:2] == dat[1:2, 0:2, 0:2]).all()
-        assert (d.section[1:2, 0:2, 0:3] == dat[1:2, 0:2, 0:3]).all()
-        assert (d.section[1:2, 0:2, 1:2] == dat[1:2, 0:2, 1:2]).all()
-        assert (d.section[1:2, 0:2, 1:3] == dat[1:2, 0:2, 1:3]).all()
-        assert (d.section[1:2, 0:2, 2:3] == dat[1:2, 0:2, 2:3]).all()
-        assert (d.section[1:2, 0:3, 0:1] == dat[1:2, 0:3, 0:1]).all()
-        assert (d.section[1:2, 0:3, 0:2] == dat[1:2, 0:3, 0:2]).all()
-        assert (d.section[1:2, 0:3, 0:3] == dat[1:2, 0:3, 0:3]).all()
-        assert (d.section[1:2, 0:3, 1:2] == dat[1:2, 0:3, 1:2]).all()
-        assert (d.section[1:2, 0:3, 1:3] == dat[1:2, 0:3, 1:3]).all()
-        assert (d.section[1:2, 0:3, 2:3] == dat[1:2, 0:3, 2:3]).all()
-        assert (d.section[1:2, 1:2, 0:1] == dat[1:2, 1:2, 0:1]).all()
-        assert (d.section[1:2, 1:2, 0:2] == dat[1:2, 1:2, 0:2]).all()
-        assert (d.section[1:2, 1:2, 0:3] == dat[1:2, 1:2, 0:3]).all()
-        assert (d.section[1:2, 1:2, 1:2] == dat[1:2, 1:2, 1:2]).all()
-        assert (d.section[1:2, 1:2, 1:3] == dat[1:2, 1:2, 1:3]).all()
-        assert (d.section[1:2, 1:2, 2:3] == dat[1:2, 1:2, 2:3]).all()
-        assert (d.section[1:2, 1:3, 0:1] == dat[1:2, 1:3, 0:1]).all()
-        assert (d.section[1:2, 1:3, 0:2] == dat[1:2, 1:3, 0:2]).all()
-        assert (d.section[1:2, 1:3, 0:3] == dat[1:2, 1:3, 0:3]).all()
-        assert (d.section[1:2, 1:3, 1:2] == dat[1:2, 1:3, 1:2]).all()
-        assert (d.section[1:2, 1:3, 1:3] == dat[1:2, 1:3, 1:3]).all()
-        assert (d.section[1:2, 1:3, 2:3] == dat[1:2, 1:3, 2:3]).all()
+        assert (d.section[:, :] == dat[:, :]).all()
+
+        for idx1 in [slice(None), 0, 1]:
```

Comment:
Please add a comment on the intent of the loops? I find it harder to read though I welcome the fix.